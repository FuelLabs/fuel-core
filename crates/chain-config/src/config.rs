mod blob;
mod chain;
mod coin;
mod consensus;
mod contract;
mod message;
#[cfg(feature = "test-helpers")]
mod randomize;
#[cfg(feature = "std")]
mod snapshot_metadata;
mod state;
mod table_entry;

pub use blob::*;
pub use chain::*;
pub use coin::*;
pub use consensus::*;
pub use contract::*;
pub use message::*;
#[cfg(feature = "test-helpers")]
pub use randomize::*;
#[cfg(feature = "std")]
pub use snapshot_metadata::*;
pub use state::*;
pub use table_entry::*;
