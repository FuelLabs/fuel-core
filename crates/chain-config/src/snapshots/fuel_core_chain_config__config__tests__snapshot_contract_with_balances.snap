---
source: crates/chain-config/src/config.rs
assertion_line: 146
expression: json
---
{
  "chain_name": "local_testnet",
  "block_gas_limit": 1000000000,
  "initial_state": {
    "contracts": [
      {
        "contract_id": "0x0000000000000000000000000000000000000000000000000000000000000000",
        "code": "0x24400000",
        "salt": "0x0000000000000000000000000000000000000000000000000000000000000000",
        "balances": [
          [
            "0x61644a25da5ad31754e23bc1eca8c766a61ceaa8c60ddc65f158a30209256e35",
            "0xe46d13fb359a76c9"
          ]
        ]
      }
    ]
  },
  "consensus_parameters": {
    "tx_params": {
      "max_inputs": 255,
      "max_outputs": 255,
      "max_witnesses": 255,
      "max_gas_per_tx": 100000000
    },
    "predicate_params": {
      "max_predicate_length": 1048576,
      "max_predicate_data_length": 1048576,
      "max_message_data_length": 1048576,
      "max_gas_per_predicate": 100000000
    },
    "script_params": {
      "max_script_length": 1048576,
      "max_script_data_length": 1048576
    },
    "contract_params": {
      "contract_max_size": 16777216,
      "max_storage_slots": 255
    },
    "fee_params": {
      "gas_price_factor": 1000000000,
      "gas_per_byte": 4
    },
    "chain_id": 0,
    "gas_costs": {
      "add": 1,
      "addi": 1,
      "aloc": 1,
      "and": 1,
      "andi": 1,
      "bal": 13,
      "bhei": 1,
      "bhsh": 1,
      "burn": 132,
      "cb": 1,
      "cfei": 1,
      "cfsi": 1,
      "croo": 16,
      "div": 1,
      "divi": 1,
      "eck1": 951,
      "ecr1": 3000,
      "ed19": 3000,
      "eq": 1,
      "exp": 1,
      "expi": 1,
      "flag": 1,
      "gm": 1,
      "gt": 1,
      "gtf": 1,
      "ji": 1,
      "jmp": 1,
      "jne": 1,
      "jnei": 1,
      "jnzi": 1,
      "jmpf": 1,
      "jmpb": 1,
      "jnzf": 1,
      "jnzb": 1,
      "jnef": 1,
      "jneb": 1,
      "k256": 11,
      "lb": 1,
      "log": 9,
      "lt": 1,
      "lw": 1,
      "mcpi": 33,
      "mint": 135,
      "mlog": 1,
      "mod": 1,
      "modi": 1,
      "move": 1,
      "movi": 1,
      "mroo": 2,
      "mul": 1,
      "muli": 1,
      "mldv": 1,
      "noop": 1,
      "not": 1,
      "or": 1,
      "ori": 1,
      "poph": 2,
      "popl": 2,
      "pshh": 2,
      "pshl": 2,
      "ret_contract": 13,
      "rvrt_contract": 13,
      "s256": 2,
      "sb": 1,
      "scwq": 13,
      "sll": 1,
      "slli": 1,
      "srl": 1,
      "srli": 1,
      "srw": 12,
      "sub": 1,
      "subi": 1,
      "sw": 1,
      "sww": 43,
      "swwq": 44,
      "time": 1,
      "tr": 105,
      "tro": 60,
      "wdcm": 1,
      "wqcm": 1,
      "wdop": 1,
      "wqop": 1,
      "wdml": 1,
      "wqml": 1,
      "wddv": 1,
      "wqdv": 2,
      "wdmd": 3,
      "wqmd": 4,
      "wdam": 2,
      "wqam": 3,
      "wdmm": 3,
      "wqmm": 3,
      "xor": 1,
      "xori": 1,
      "call": {
        "base": 144,
        "dep_per_unit": 214
      },
      "ccp": {
        "base": 15,
        "dep_per_unit": 103
      },
      "csiz": {
        "base": 17,
        "dep_per_unit": 790
      },
      "ldc": {
        "base": 15,
        "dep_per_unit": 272
      },
      "logd": {
        "base": 26,
        "dep_per_unit": 64
      },
      "mcl": {
        "base": 1,
        "dep_per_unit": 3333
      },
      "mcli": {
        "base": 1,
        "dep_per_unit": 3333
      },
      "mcp": {
        "base": 1,
        "dep_per_unit": 2000
      },
      "meq": {
        "base": 1,
        "dep_per_unit": 2500
      },
      "retd_contract": {
        "base": 29,
        "dep_per_unit": 62
      },
      "smo": {
        "base": 209,
        "dep_per_unit": 55
      },
      "srwq": {
        "base": 47,
        "dep_per_unit": 5
      }
    }
  },
  "consensus": {
    "PoA": {
      "signing_key": "22ec92c3105c942a6640bdc4e4907286ec4728e8cfc0d8ac59aad4d8e1ccaefb"
    }
  }
}
